kibana-reports/public/components/context_menu/context_menu.js

```diff
@@ -0,0 +1,234 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+ 
+ /* eslint-disable no-restricted-globals */
+ import $ from 'jquery';
+ import dateMath from '@elastic/datemath';
+ import { readStreamToFile } from '../main/main_utils';
+ import { 
+  contextMenuCreateReportDefinition,
+  getTimeFieldsFromUrl,
+  displayLoadingModal,
+  addSuccessOrFailureToast
+} from './context_menu_helpers';
+import { 
+  popoverMenu,
+  getMenuItem, 
+} from './context_menu_ui';
+
+ const replaceQueryURL = () => {
+  let url = window.location.href;
+  let timeString = url.substring(
+    url.lastIndexOf("time:"),
+    url.lastIndexOf("))")
+  );
+
+  let fromDateString = timeString.substring(
+    timeString.lastIndexOf("from:") + 5,
+    timeString.lastIndexOf(",")
+  );
+
+  fromDateString = fromDateString.replace(/[']+/g, '');
+  let fromDateFormat = dateMath.parse(fromDateString);
+
+  let toDateString = timeString.substring(
+    timeString.lastIndexOf("to:") + 3,
+    timeString.length
+  );
+  toDateString = toDateString.replace(/[']+/g, '');
+  let toDateFormat = dateMath.parse(toDateString);
+
+  // replace to and from dates with absolute date
+
+  url = url.replace(
+    fromDateString + "))",
+    '\'' + fromDateFormat.toISOString() + "\'"
+  );
+
+  url = url.replace(
+    toDateString + "))", 
+    '\'' + toDateFormat.toISOString() + "\'))"
+  );
+  return url;
+ }
+
+ const generateInContextReport = (timeRanges, queryUrl, fileFormat) => {
+  displayLoadingModal();
+  let baseUrl = window.location.href.substr(0, window.location.href.indexOf('?'));
+  let reportSource = "";
+  if (window.location.href.includes('dashboard')) {
+    reportSource = "Dashboard";
+  }
+  else if (window.location.href.includes('visualize')) {
+    reportSource = "Visualization";
+  }
+
+  // create query body
+  let contextMenuOnDemandReport = {
+    query_url: queryUrl,
+    time_from: timeRanges.time_from.valueOf(),
+    time_to: timeRanges.time_to.valueOf(),
+    report_definition: {
+      report_params: {
+        report_name: "On_demand_report",
+        report_source: reportSource,
+        description: "In-context report download",
+        core_params: {
+          base_url: baseUrl,
+          report_format: fileFormat,
+          time_duration: timeRanges.time_duration
+        }
+      },
+      trigger: {
+        trigger_type: "On demand"
+      }
+    }
+  };
+
+  fetch('/api/reporting/generateReport', {
+    headers: {
+      "Content-Type": "application/json",
+      "kbn-version": "7.8.0",
```

Comment:
Does this needs update as repo is migrated to 7.9.1?

Reply:
Good catch- I think it may be safe to remove this field entirely to avoid potential issue as well

Reply:
So looks like it breaks without this line- but since this PR still is in a `7.8` version I don't think we should update here because Zhongnan's changes are not merged in yet